web/source/keyboards/kmwkeyboards.ts

```diff
@@ -126,7 +135,7 @@ namespace com.keyman.keyboards {
      */
     registrationResolvers: {[timeoutID: number] : RegistrationPromiseTuple} = {};
 
-    languageListPromise: Promise<KeyboardStub[]> = null;
+    languageListPromise: Promise<(KeyboardStub)[]> = null;
```

Comment:
If the PR remains otherwise remains in its current state, this needs to be `Promise<KeyboardStub | ErrorStub>`, as `ErrorStub`s can still be provided to the `resolve` function and thus the `then` clause.